docs/operations-guide/encrypting-database-details-at-rest.md

```diff
@@ -1,21 +1,51 @@
+recommend using a secure random key generator to generate it. `openssl` is a
+good choice:
 
     openssl rand -base64 32
 
-This gives you a cryptographically-secure, randomly-generated 32-character key that will look something like `IYqrSi5QDthvFWe4/WdAxhnra5DZC3RKx3ZSrOJDKsM=`. Set it as an environment variable and
-start Metabase as usual:
+This gives you a cryptographically-secure, randomly-generated 32-character key
+that will look something like `IYqrSi5QDthvFWe4/WdAxhnra5DZC3RKx3ZSrOJDKsM=`.
+Set it as an environment variable and start Metabase as usual:
 
     MB_ENCRYPTION_SECRET_KEY="IYqrSi5QDthvFWe4/WdAxhnra5DZC3RKx3ZSrOJDKsM=" java -jar metabase.jar
 
-**Note** Some versions of Linux interpret single-quotes (`'`) and double-quotes (`"`) differently for environment variable values, so if you upgrade to a newer version of Linux, and the key originally used single-quotes, you might need to wrap the key in double-quotes, so that the single-quotes are interpreted literally. For example, `MB_ENCRYPTION_SECRET_KEY='IYq...sM='` would be `MB_ENCRYPTION_SECRET_KEY="'IYq...sM='"`
+**Note** Some versions of Linux interpret single-quotes (`'`) and double-quotes
+(`"`) differently for environment variable values, so if you upgrade to a newer
+version of Linux, and the key originally used single-quotes, you might need to
+wrap the key in double-quotes, so that the single-quotes are interpreted
+literally. For example, `MB_ENCRYPTION_SECRET_KEY='IYq...sM='` would be
+`MB_ENCRYPTION_SECRET_KEY="'IYq...sM='"`
 
-Metabase will securely encrypt and store the connection details for any new Databases you add. (Connection details for existing databases will be encrypted as well if you save them in the admin panel).
-Existing databases with unencrypted details will continue to work normally.
+Metabase will securely encrypt and store the connection details for any new
+Databases you add. (Connection details for existing databases will be encrypted
+as well if you save them in the admin panel). Existing databases with
+unencrypted details will continue to work normally.
 
 >Take care not to lose this key because you can't decrypt connection details without it. If you lose (or change) the key, you'll have to reset all of the connection details that have been encrypted with it in the Admin Panel.
+
+# Key rotation
+
+Starting on version 0.38, you can rotate the encryption key using the cli
+command `rotate-encryption-key`. This command needs both the current encryption
+key set as `MB_ENCRYPTION_SECRET_KEY` and the new encryption key as a parameter.
+
+`MB_ENCRYPTION_SECRET_KEY=your-current-key java -jar metabase.jar rotate-encryption-key new-key`
+
+Metabase app should be stopped while running this command, and restarted with
+the new key as the `MB_ENCRYPTION_SECRET_KEY` environment variable.
+
+Note: As with all data migrations, it's recommended that you
+[backup](./backing-up-metabase-application-data.html) your data before doing a
```

Comment:
MD-file links are automatically converted, when it's added to the website, but keep the MD-files here, so the docs also function on the GitHub (in case someone reads it there).
```suggestion
[backup](./backing-up-metabase-application-data.md) your data before doing a
```

Reply:
woah, didn't know it was that smart!